classes/navigationMenu/NavigationMenuDAO.inc.php

```diff
@@ -314,42 +314,26 @@ function installSettings($contextId, $filename) {
 	 * @param int $id
 	 */
 	function unCache($id){
-		$cache = $this->getCache($id);
-		if ($cache) $cache->flush();
+		$item = $this->_getCache($id);
+		$item->clear();
 	}
 
 	/**
 	 * Get the settings cache for a given ID
 	 * @param $id
-	 * @return array|null (Null indicates caching disabled)
+	 * @return Stash\Item
 	 */
-	function getCache($id) {
-		static $navigationMenuCache;
-		if (!isset($navigationMenuCache)) {
-			$navigationMenuCache = array();
+	protected function _getCache($id) {
+		$pool = new Stash\Pool(new Stash\Driver\FileSystem(array('path' => Core::getFileCachePath() . '/stash')));
+		$item = $pool->getItem('navigationMenu' . md5($id));
```

Comment:
What benefit do we get from calling `md5` on the `id`?

Reply:
None -- I've removed it.